ROADMAP.md

```diff
@@ -0,0 +1,187 @@
+# New Differentiation Tools that Reduce User Effort
+
+**Motivation:** While analytic derivatives provide massive performance gains, they also require 
+significant user effort to implement. This creates a high activation energy that prevents many 
+users from taking advantage of the most powerful feature of OpenMDAO. 
+
+**Overall Goal:** Lower the effort required to implement analytic partial derivatives for components, 
+offering a spectrum of options that can potentially trade required user effort with computational efficiency. 
+
+## 1) Coloring applied to approximated partial derivatives 
+### Goal: 
+- Make coloring usable for FD and CS approximations.
+- Offer much higher performance for components with very sparse partial derivative Jacobians 
+  (e.g. vectorized components with no interaction between vector elements).
+- When used with CS, this effectively offers a fast-forward mode AD.
+
+### Potential Challenges: 
+- The computational cost of coloring may be excessive for models with a lot of instances.
+- Coloring algorithms are not perfect and may result in an invalid Jacobian, 
+  so we need an effective way for users to check their colored partials. 
+  The existing `check_partials` functionality can be leveraged, but may need some updates.
+- In some cases, sparsity may not be high enough to offer meaningful performance gains. 
+  We need to provide a means for the user to check this.
+
+### Notes: 
+- A prototype implementation is already available as an experimental feature.
+- The partial coloring algorithm will also be useful for AD partials.
+
+
+## 2) Algorithmic differentiation for component partial derivatives
+    
+### Goal: 
+- Provide forward and reverse mode AD that works for a wide variety of general use cases including
+  with many numpy functions.
+- AD should have relatively good performance compared to hand differentiation.
+
+### Potential Challenges: 
+- AD tools will struggle with current OpenMDAO syntax for `compute` and `apply_nonlinear` methods, 
+  so some kind of translation layer will be needed.
+- Python AD libraries are not as well developed as those for other languages (e.g. C, Fortran, Julia).
+- Certain coding practices are not compatible with AD, and will need to be avoided
+  (e.g. modification of instance attributes, functions with side-effects).
+
+
+----------------------------------------------
+----------------------------------------------
+----------------------------------------------
+# Establishing A Community Contribution Process
+
+## 1) Planning for the 2020 OpenMDAO workshop 
+
+## 2) POEMs
+- A POEM is a **p**roposal for **O**penMDAO **e**nhance**m**ent.
+- This process is loosely based on the model used by cPython project (the PEP process).
+- A new repository has been created for tracking POEMs (http://github.com/openmdao/POEMs).
+- We are planning to work the POEM process on PR 1086 (https://github.com/OpenMDAO/OpenMDAO/pull/1086).
+
```

Comment:
Do we need to work a poem on something that is destined to be a plugin? (or perhaps it isn't)